examples/mlperf/initializers.py

```diff
@@ -41,7 +41,8 @@ def __init__(self, in_features, out_features, bias=True, std=0.02):
     self.bias = Tensor.zeros(out_features, dtype=dtypes.float32) if bias else None
   
   def __call__(self, x:Tensor):
-    return x.linear(self.weight.cast(dtypes.default_float).transpose(), self.bias.cast(dtypes.default_float) if self.bias is not None else None)
+    matmul_dtype = dtypes.half if getenv("HALF_LINEAR", 0) else dtypes.default_float
+    return x.contiguous().cast(matmul_dtype).linear(self.weight.contiguous().cast(matmul_dtype).transpose(), self.bias.cast(dtypes.default_float) if self.bias is not None else None)
```

Comment:
do you need these `contiguous`? weight should be contiguous and is there a benefit to do this with `x`?

Reply:
currently, uop count explodes when I don't do that. Seems like a bug. I think Qazalin is on that already

Reply:
if it's a temporary issue, can you add a TODO note for some context

Reply:
Yeah I'm looking into the root cause for this